### PR TITLE
user can configure one or multiple ec2 instances for Controller and Hub

### PR DESCRIPTION
**Pull request by qjones7**
This PR implements the creation and configuration of one or multiple ec2 instances for controller and hub.

JIRA issue for reference: [https://issues.redhat.com/browse/AAP-18596](https://issues.redhat.com/browse/AAP-18596)

Steps to test:

Pull down the pr
Note the main vars (number of instances) variables for controller and hub added in PR. The number is set to 2 for testing but i can set to 1 prior to merge 
Deploy resources using steps mentioned here [(https://github.com/ansible-content-lab/aws_terraform_deployment#deploying-infrastructure](https://github.com/ansible-content-lab/aws_terraform_deployment#deploying-infrastructure)

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -98,3 +98,15 @@ variable "infrastructure_execution_count" {
   type = number
   default = 1
 }
+
+variable "number_of_controller_instances" {
```
Comment (chrisahl) [inline variables.tf]: I named these slightly different for eda and execution based on the way the example I had looked at was naming stuff. We will need to come to a consensus and be consistent.

File: variables.tf [hunk 2]
```diff
@@ -98,3 +98,15 @@ variable "infrastructure_execution_count" {
...
+  description = "The number of ec2 instances for controller"
+  type = number
+  default = 2
+}
+variable "number_of_hub_instances" {
```
Comment (chrisahl) [inline variables.tf]: I named these slightly different for eda and execution based on the way the example I had looked at was naming stuff. We will need to come to a consensus and be consistent.

File: main.tf [hunk 3]
```diff
@@ -60,6 +60,7 @@ data "aws_ami" "instance_ami" {
 module "controller_vm" {
   source = "./modules/vms"
 
+  count = var.number_of_controller_instances
   deployment_id = var.deployment_id
   instance_name_suffix = random_string.instance_name_suffix.result
   vm_name_prefix = "controller-"
```
Comment (chrisahl) [inline main.tf]: for vm_name_prefix change to "controller-${count.index + 1}-"

File: main.tf [hunk 4]
```diff
@@ -69,6 +70,19 @@ module "controller_vm" {
   subnet_id = module.vpc.infrastructure_subnets[0]
 }
 
+module "hub_vm" {
+  source = "./modules/vms"
+
+  count = var.number_of_hub_instances
+  deployment_id = var.deployment_id
+  instance_name_suffix = random_string.instance_name_suffix.result
+  vm_name_prefix = "hub-"
```
Comment (chrisahl) [inline main.tf]: for vm_name_prefix change to "hub-${count.index + 1}-"

File: main.tf [hunk 5]
```diff
@@ -69,6 +70,19 @@ module "controller_vm" {
...
+  # desired ami id can be specified by replacing below line with `instance_ami = <desired-ami-id-here>`
+  instance_ami = data.aws_ami.instance_ami.id
+  vpc_security_group_ids = [module.vpc.infrastructure_sg_id]
+  subnet_id = module.vpc.infrastructure_subnets[0]
```
Comment (chrisahl) [inline main.tf]: This should be _subnets[2] based on how variable "infrastructure_vpc_subnets" is defined

File: variables.tf [hunk 6]
```diff
@@ -98,3 +98,15 @@ variable "infrastructure_execution_count" {
...
+  description = "The number of ec2 instances for hub"
+  type = number
+  default = 2
```
Comment (chrisahl) [inline variables.tf]: Assume we want to default to 1 hub

**Review by chrisahl — CHANGES_REQUESTED** [review]
comments inline

**Comment by nathanweatherly** [comment]
JIRA issue for reference: https://issues.redhat.com/browse/AAP-18596

File: variables.tf (hunk 6 above)
Comment (qjones7) [reply variables.tf]: corrected,  had it that way for PR testing

File: variables.tf (hunk 2 above)
Comment (qjones7) [reply variables.tf]: complete

File: variables.tf (hunk 1 above)
Comment (qjones7) [reply variables.tf]: complete

File: main.tf (hunk 5 above)
Comment (qjones7) [reply main.tf]: complete

File: main.tf (hunk 4 above)
Comment (qjones7) [reply main.tf]: complete

File: main.tf (hunk 3 above)
Comment (qjones7) [reply main.tf]: complete

File: variables.tf [hunk 7]
```diff
@@ -98,3 +98,15 @@ variable "infrastructure_execution_count" {
   type = number
   default = 1
 }
+
+variable "controller_instances_count" {
```
Comment (chrisahl) [inline variables.tf]: Change to infrastructure_controller_count to align with existing variables

File: variables.tf [hunk 8]
```diff
@@ -98,3 +98,15 @@ variable "infrastructure_execution_count" {
...
+  description = "The number of ec2 instances for controller"
+  type = number
+  default = 1
+}
+variable "hub_instances_count" {
```
Comment (chrisahl) [inline variables.tf]: Change to infrastructure_hub_count to align with existing variables

File: variables.tf (hunk 7 above)
Comment (qjones7) [reply variables.tf]: I totally missed what you were saying. I made the correction

File: variables.tf (hunk 8 above)
Comment (qjones7) [reply variables.tf]: complete

**Review by chrisahl — APPROVED** [review]

**Pull request merged**